backend/tournesol/tests/test_text_search.py

```diff
@@ -0,0 +1,712 @@
+
+
+    def test_stop_words(self):
+        """
+        The words that carry no meaning ("stop words") should be
+        automatically removed by Postgres. In a language-dependent way.
+        """
+        query = "cake and so my the he again been"
+
+        self._create_rated_entity(field1="and so my the he again been")
+
+        response = self.client.get(
+            self._make_url(query),
+            format="json"
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data["count"], self.setup_results_count)
+
+        self._create_rated_entity(field1="cake")
+
+        response = self.client.get(
+            self._make_url(query),
+            format="json"
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data["count"], 1 + self.setup_results_count)
+
+    def test_stop_words_french(self):
+        """
+        Each language has its stop words dictionary.
+        Check that it works for french too.
+        """
+        query = "oignon le ma elle du où"
+
+        self._create_rated_entity(field1="le ma elle du où", language="fr")
+
+        response = self.client.get(
+            self._make_url(query, "fr"),
+            format="json"
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data["count"], 0)
+
+        self._create_rated_entity(field1="oignon", language="fr")
+
+        response = self.client.get(
+            self._make_url(query, "fr"),
+            format="json"
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data["count"], 1)
+
+
+    def test_stemming(self):
+        """
+        Only the root of words is indexed or queried.
+        This allows to match other words that have the same root,
+        and thus usually the same meaning.
+        """
+        word = "fruitful"
+        query = "fruits"
+        self._create_rated_entity(field1=word)
+
+        response = self.client.get(
+            self._make_url(query),
+            format="json"
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data["count"], self.setup_results_count + 1)
+
+    def test_stemming_french(self):
+        """
+        Each language has its stemming dictionary.
+        Check that it works for french too.
+        """
+        word = "construction"
+        query = "constructif"
+        self._create_rated_entity(field1=word, language="fr")
+
+        response = self.client.get(
+            self._make_url(query, "fr"),
+            format="json"
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data["count"], 1)
+
+    def test_languages_filter(self):
+        """
+        There should be a filter on the entities' languages

```

Comment:
Given the fact it's not explicit here that `self.url` contains the `en` language filter, it's not easy to understand why the `fr` entities are not returned.

I suggest to make it explicit that `fr` entities don't match the query languages.

```suggestion
        Only entities matching the query languages must be returned.
```